libp2p/protocols/connectivity/autonat/client.nim

```diff
@@ -61,14 +67,36 @@ method dialMe*(
     incomingConnection.cancel()
       # Safer to always try to cancel cause we aren't sure if the peer dialled us or not
     if incomingConnection.completed():
-      await (await incomingConnection).connection.close()
-  trace "sending Dial", addrs = switch.peerInfo.addrs
-  await conn.sendDial(switch.peerInfo.peerId, switch.peerInfo.addrs)
-  let response = getResponseOrRaise(AutonatMsg.decode(await conn.readLp(1024)))
+      try:
+        await (await incomingConnection).connection.close()
+      except:
+        discard
```

Comment:
Curious about the exception being generated here. Do you think it's worth it to add a trace log?

Reply:
i internationally didn't wanted to add log there, as compiler tells that `AlreadyExpectingConnectionError` can happen here, but, if this error happens it will happen only right after `switch.connManager.expectConnection(pid, In)` is called and that case is already handled. however i have added comment and `error` log. it should never happen tho